src/test/rgw/test_rgw_string.cc

```diff
@@ -12,6 +12,7 @@
  *
  */
 
+#include <errno.h>
```

Comment:
okay, yeah

> Include files that require other include files to be included first.

i would guess that most of those cases arise by accident, rather than by design. the google c++ style guide is pretty explicit about this, putting the section about [Self-contained headers](https://google.github.io/styleguide/cppguide.html#Self_contained_Headers) at the very top